### PR TITLE
Fix INVERT_BNDARY_ONE case for PCR solver

### PR DESCRIPTION
**Pull request by JosephThomasParker**
This fixes cases where we use the PCR solver with boundary conditions that use the `INVERT_BNDRY_ONE` flag.

As input, the PCR solver takes a tridiagonal matrix with number of rows/columns that is a power of 2. In BOUT++ test cases, the problem domain is typically a power of 2 plus some guard cells. In the original implementation, we did a row elimination on the first and last non-guard cells so that we could decouple the 2^n-sized problem domain and pass it to the solver.

In some tests cases however, `INVERT_BNDRY_ONE` is set while also specifying that there are two guard cells. In such cases, the problem domain actually as 2^n + 2 points, so the solver fails.

This PR introduces a workaround for those cases where we do forward elimination across all lower boundary cells and backwards elimination across all upper boundary cells to decouple the middle 2^n points to give to the PCR solver.

Schematically:
**Old working case**
```
                      boundary. | interior
boundary ->               x. x. |  0  0. 0
boundary ->.              0  x. |  x. 0. 0
interior ->               0. x. |. x. x. 0
interior ->.              0. 0. |  x. x. x   
```
becomes
```
                      boundary. | interior
boundary ->               x. x. |  0  0. 0
boundary ->.              0  x. |  x. 0. 0
interior ->               0. 0. |. Y. x. 0
interior ->.              0. 0. |  x. x. x   
```
after doing forward elimination from row 1 to row 2.  The interior rows are decoupled and we can use PCR.

**Case where `mxg=2` but `INVERT_BNDRY_ONE` is set**
```
                      boundary. | interior
boundary ->               x. x. |  0  0. 0
boundary ->.              x. x. |  x. 0. 0     <---- this row has three elements - it's really an internal row
interior ->               0. x. |. x. x. 0
interior ->.              0. 0. |  x. x. x   
```
But we can reduce this to the same form as above by doing forward elimination from row 0 to row 1, and then from row 1 to row 2:
```
                      boundary. | interior
boundary ->               x. x. |  0  0. 0
boundary ->.              0  Y. |  x. 0. 0
interior ->               0. 0. |. Z. x. 0
interior ->.              0. 0. |  x. x. x   
```

### CODE REVIEW

**Comment by JosephThomasParker** [comment]
Oh right, addresses #2350, at least for the `test-gyro` case.

File: src/invert/laplace/impls/pcr/pcr.hxx [hunk 1]
```diff
@@ -118,6 +118,11 @@ private:
   /// MPI process x ID
   int xproc;
 
+  /// Number of inner boundary cells
+  int inbndry;
+  /// Number of outer boundary cells
+  int outbndry;
```
Comment (ZedThree) [inline src/invert/laplace/impls/pcr/pcr.hxx:124]: Is this change necessary? Looks like they could be kept as local variables?

**Review by ZedThree — APPROVED** [review]
Confirmed fix for `test-gyro`, but not enough to fix the PETSc Laplace tests

**Comment by ZedThree** [comment]
I guess there's still an issue with `INVERT_AC_GRAD`
